account_balance_query.go

```diff
@@ -76,52 +67,50 @@ func (query *AccountBalanceQuery) validateNetworkOnIDs(client *Client) error {
 	return nil
 }
 
-func (query *AccountBalanceQuery) build() *AccountBalanceQuery {
+func (query *AccountBalanceQuery) build() *proto.Query_CryptogetAccountBalance {
+	pb := proto.CryptoGetAccountBalanceQuery{Header: &proto.QueryHeader{}}
+
 	if !query.accountID.isZero() {
-		query.pb.BalanceSource = &proto.CryptoGetAccountBalanceQuery_AccountID{
+		pb.BalanceSource = &proto.CryptoGetAccountBalanceQuery_AccountID{
 			AccountID: query.accountID.toProtobuf(),
 		}
 	}
 
 	if !query.contractID.isZero() {
-		query.pb.BalanceSource = &proto.CryptoGetAccountBalanceQuery_ContractID{
+		pb.BalanceSource = &proto.CryptoGetAccountBalanceQuery_ContractID{
 			ContractID: query.contractID.toProtobuf(),
 		}
 	}
 
-	return query
+	return &proto.Query_CryptogetAccountBalance{
+		CryptogetAccountBalance: &pb,
+	}
+
 }
 
 func (query *AccountBalanceQuery) GetCost(client *Client) (Hbar, error) {
 	if client == nil || client.operator == nil {
 		return Hbar{}, errNoClientProvided
 	}
+	query.nodeIDs = client.network.getNodeAccountIDsForExecute()
 
-	paymentTransaction, err := query_makePaymentTransaction(TransactionID{}, AccountID{}, client.operator, Hbar{})
+	err := query.validateNetworkOnIDs(client)
 	if err != nil {
 		return Hbar{}, err
 	}
 
-	query.pbHeader.Payment = paymentTransaction
-	query.pbHeader.ResponseType = proto.ResponseType_COST_ANSWER
-	if len(query.Query.GetNodeAccountIDs()) == 0 {
-		query.SetNodeAccountIDs(client.network.getNodeAccountIDsForExecute())
-	}
-
-	err = query.validateNetworkOnIDs(client)
+	protoReq, err := query.costQueryMakeRequest(client)
 	if err != nil {
 		return Hbar{}, err
 	}
 
-	query.build()
-
 	resp, err := execute(
 		client,
 		request{
 			query: &query.Query,
 		},
 		accountBalanceQuery_shouldRetry,
-		costQuery_makeRequest,
```

Comment:
Shouldn't this be `costQueryMakeRequest`?

Reply:
was  but confused what you meant but, 
`protoReq, err := query.costQueryMakeRequest(client)`

it returns an error so you have to catch it, right